src/transform/src/join_implementation.rs

```diff
@@ -207,7 +211,10 @@ mod delta_queries {
                 &input_relation[..],
                 prior_arities,
             );
-            if !orders.iter().all(|o| o.iter().all(|(c, _, _)| c.arranged)) {
+            if !orders
+                .iter()
+                .all(|o| o.iter().skip(1).all(|(c, _, _)| c.arranged))
+            {
```

Comment:
I'm not clear on why we skip one of the results of this method. While it is fresh in you mind, could you leave a comment explaining the logic?